css: rebirth of the bounty

thelostone-mc opened the pull request:
##### Description
<!-- A description on what this PR aims to solve -->
![x](https://user-images.githubusercontent.com/5358146/37836635-fb344472-2ed8-11e8-9160-efe7237b6681.gif)


##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, ui, crypto, etc). -->
- css + js + html

##### Refers/Fixes
- https://github.com/gitcoinco/web/issues/419
<!--
  Link to an issue if applicable. For example:
  If your PR fixes an issue  -> Fixes: #102
  If your PR refers an issue -> Refs: #101
-->

owocki commented:
very exciting!  is this ready for some serious QA yet?  might be worth putting up on a staging so the community can test it

mkosowsk commented:
😍😍😍

mbeacom commented:
@owocki @thelostone-mc This branch is deployed to https://stage.gitcoin.co

mbeacom commented on app/assets/v2/js/pages/bounty_details.js:
```diff
@@ -467,8 +541,8 @@ var pull_bounty_from_api = function() {
         do_actions(result);
-
-        render_fulfillments(result);
+        
```
And here.

mbeacom commented on app/assets/v2/js/pages/bounty_details.js:
```diff
@@ -301,7 +370,7 @@ var build_detail_page = function(result) {
     var head = null;
     var val = result[key];
-
+    
```
And here. 😛

owocki commented:
from https://stage.gitcoin.co/funding/details/?url=https://github.com/owocki/adblock-to-bitcoin/issues/5:

http://bits.owocki.com/263C1H2c3f21/Screen%20Shot%202018-03-23%20at%2010.09.37%20AM.png

tooltips seem borked

should the tags be clickable and send you back to the explorer?

http://bits.owocki.com/200A1I0b2605/Screen%20Shot%202018-03-23%20at%2010.10.48%20AM.png

my bounty is in an open state, it should say 'none' under each of these contributor areas.  its confusing because at first glance , i think that itss both work started, work submitted, and accepted status

http://bits.owocki.com/0j1S1q1J2L3f/Screen%20Shot%202018-03-23%20at%2010.11.29%20AM.png

all activity should say 'none' under it 

http://bits.owocki.com/213n3C0S1r3Z/Screen%20Shot%202018-03-23%20at%2010.11.58%20AM.png

gitcoin profile is not showing up: 

http://bits.owocki.com/3u1a1o0n382A/Screen%20Shot%202018-03-23%20at%2010.12.23%20AM.png

thelostone-mc commented:
@mbeacom 
- removed the trailing white space 
- fixed the indentation (This was cause I decided to try out vscode against atom. forgot to override default setting :P )

mbeacom commented:
@thelostone-mc 👍 for vscode - It should be down to the remaining issues from Kevin's comment.

thelostone-mc commented:
@owocki 
> should the tags be clickable and send you back to the explorer?
 
It would make sense if our explorer can filter out issues based on the keyword. Does the explorer support it as of this moment ? 

> tooltips seem borked

No no it's not broken,  I just forgot to add the content. derp! Then again you could call that broken :P 

> gitcoin profile is not showing up:

The reason for this is cause `bounty_owner_github_username` from the API at times is empty
@mbeacom / @owocki  could you tell me which property to use instead ?

<img width="1440" alt="screen shot 2018-03-23 at 11 24 49 pm" src="https://user-images.githubusercontent.com/5358146/37845729-6b87d622-2ef1-11e8-8e72-8009817dfe43.png">

will address the display none text (@PixelantDesign did say that -> sort of forgot about it)

Also need to see why the comments aren't showing up

thelostone-mc commented:
Things left to do : 
- Add the fancy time left bar 
- On Click Start Work -> Popup to let you post comment to github

^Will shoot these out as a separate PR later on ^_^

thelostone-mc commented:
>yes. gitcoin.co/explorer?q=KEYWORD
@owocki the tags are not links to the explorer :) Thanks for that

@mbeacom am i missing out anything obvious ?

thelostone-mc commented:
@mbeacom @vs77bb @owocki @mkosowsk @PixelantDesign 
Would it be alright if I removed the box for the funder section ? It's always gonna be a single funder -> so it seems redundant ! Your thoughts?

thelostone-mc commented:
@owocki @PixelantDesign I've added a success notification on start / stop work
(I know the huge banner is meh. I just reused what's available )

<img width="1440" alt="screen shot 2018-03-24 at 3 58 26 pm" src="https://user-images.githubusercontent.com/5358146/37862995-8d5b28ea-2f7c-11e8-8481-8b79cac51b96.png">

@PixelantDesign wanted : 

<img width="686" alt="screen shot 2018-03-24 at 4 02 17 pm" src="https://user-images.githubusercontent.com/5358146/37863013-c55817bc-2f7c-11e8-8e40-8d6f5ba82b34.png">

Would you happen to know if this small notif is already implemented ! ( we need a styleguide :P )

PixelantDesign commented:
looking great! No the smalker one has not been implemented. I also think changing the height of the confirmation notification would be fine for now.

PixelantDesign commented:
@thelostone-mc, @owocki 
I'm okay with this approach:

<img width="1440" alt="37849470-ddbb6b3a-2efd-11e8-933d-d8ad30f9ea8d" src="https://user-images.githubusercontent.com/238235/37863858-d9bbf692-2f3b-11e8-972a-f154c7a87318.png">

mkosowsk commented:
> Would it be alright if I removed the box for the funder section ? It's always gonna be a single funder -> so it seems redundant ! Your thoughts?

@thelostone-mc think this approach is fine for now but I do believe it's in the roadmap to have multiple funders in the future 🤔so might be needed to be added back at a later date (should be painless tho)

thelostone-mc commented:
@mbeacom updated stuff. left a few questions open regarding the API response in https://github.com/gitcoinco/web/pull/691#issuecomment-375750920

owocki commented:
can we deploy the latest to stage for another round of testing?!?

mbeacom commented:
@owocki This is back up on https://stage.gitcoin.co

owocki commented:
testing on stage now.. looks very good!

notes: 
1. i put a fix in on master for the issue wehre the submitters github username was not showing up on the bounty 3a2d97ce8f48036a6b10be4c8a2cd89750ae7580
2. it looks like the 'opened' time resets after you submit work ( http://bits.owocki.com/2W243R1Q0112/Screen%20Shot%202018-03-26%20at%208.48.43%20AM.png ) -- is this field populated with web3_created or with `created_on` from the API?  it should be the former
2. similarly, the timestamps on work submitted and accepted seem to all default to the latest action's timestamp: http://bits.owocki.com/3f0r1I1G1O15/Screen%20Shot%202018-03-26%20at%208.51.09%20AM.png
3. responsive design looks good!
4. looks like theres some CI fixes to make from travis.

lets get this shipped soon!

owocki commented:
ux question: if i start work and then stop work, should the feed (1) show two entries (one for each action), or should (2) the 'stop work' activity and the startwork activty combine together and then destroy each other, and the feed appears like it never happened?

thelostone-mc commented:
@owocki Ideally we shouldn't as it maintains hisotry (I sort of followed the old implementation though )

thelostone-mc commented:
> it looks like the 'opened' time resets after you submit work ( http://bits.owocki.com/2W243R1Q0112/Screen%20Shot%202018-03-26%20at%208.48.43%20AM.png ) -- is this field populated with web3_created or with created_on from the API? it should be the former

@owocki Ah okay ! Fixed this.

> the timestamps on work submitted and accepted seem to all default to the latest action's timestamp

I'm using the only timestamp available for this from the API response

For accepted / submmited : [L578](https://github.com/thelostone-mc/web/blob/d1da9446cbde19c9c99aff890920b84de16436dc/app/assets/v2/js/pages/bounty_details.js#L578), [L589](https://github.com/thelostone-mc/web/blob/d1da9446cbde19c9c99aff890920b84de16436dc/app/assets/v2/js/pages/bounty_details.js#L586)

For started: [L597](https://github.com/thelostone-mc/web/blob/d1da9446cbde19c9c99aff890920b84de16436dc/app/assets/v2/js/pages/bounty_details.js#L597)

owocki commented:
can we integrate the 'increase payout' button that was added in at https://github.com/gitcoinco/web/pull/626#issuecomment-376214588 into the new bounty detail page?

owocki commented:
i will look into the timestamp stuff above now

owocki commented:
here is an upstream fix for the timestamp stuff => 1477222fac5ef9479e1fc136818122ece55debcf

thelostone-mc commented:
@mbeacom @owocki added :) 
<img width="1440" alt="screen shot 2018-03-26 at 11 49 59 pm" src="https://user-images.githubusercontent.com/5358146/37924764-75f29348-3150-11e8-9367-df027446295c.png">

owocki commented:
@mbeacom @thelostone-mc i'm thikning we should ship this soon.  what do you think?

PixelantDesign commented:
<img width="1453" alt="screen shot 2018-03-26 at 3 06 52 pm" src="https://user-images.githubusercontent.com/238235/37927196-694dde20-3107-11e8-8226-edf6bdc17c63.png">

@thelostone-mc user should not be able to submit work until they've started work. Please hide submit work button.

PixelantDesign commented:
@thelostone-mc 
<img width="1418" alt="screen shot 2018-03-26 at 3 34 15 pm" src="https://user-images.githubusercontent.com/238235/37928565-3ac9db90-310b-11e8-913d-8da2d0202ae7.png">

owocki commented:
> The activity feed is intended to capture all of the activity that has happened to date as it relates to the progression of the ticket from a work state perspective. If possible, activities should not cancel each other out.


@thelostone-mc can this be pulled into a follow on PR along with the start work modal?

thelostone-mc commented:
> user should not be able to submit work until they've started work. Please hide submit work button.

@PixelantDesign @owocki I was able to directly submit stuff without clicking on start work. Are we sure we don't want this ?


> @thelostone-mc can this be pulled into a follow on PR along with the start work modal?
@owocki aye aye captain

@PixelantDesign hide if user doesn't have a profile would be resolved once the API is fixed, are you sure you want a temp fix for that ?

Also the lineup  -> you are talking about the All Activity & Funder section right ?

PixelantDesign commented:
@thelostone-mc 
Can you explain a use case where a user is submitting work for an issue that they've not committed to?

> Also the lineup -> you are talking about the All Activity & Funder section right ?

Line up profile icons.

Update tooltips for each button state:

**View on Github**: View issue details and comments on Github
**Start Work**: Notify the funder that you would like to take on this project
**Submit Work**: Submit work for the funder to review
**Cancel Bounty**: Cancel bounty and reclaim funds for this issue

Thank you.

thelostone-mc commented:
@PixelantDesign 
I always liked having the submit button readily available 
Start work is sort of just calling it out to folks saying hi I’m working on this  (I liked it that it was optional )
If you think we should make it compulsory, then I'm okay with it 


> Line up profile icons. 

Done!!

> Thank you.

Hold off the thanks until I push in the changes :P

darkdarkdragon commented:
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__This issue now has a funding of 0.001 ETH (0.69 USD @ $692.1/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](http://localhost:8000/issue/gitcoinco/web/691/539).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $0 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
